---
title: "Assignment 2"
author: "Mouaid Alim"
date: ""
output: 
  html_document:
    fig_caption: yes
    toc: true
    toc_depth: 3
bibliography: references.bib
editor_options: 
  markdown: 
    wrap: 72
---

# Background

### Data and Study Objective

In the first assignment, the focus was on analyzing a dataset from the
Gene Expression Omnibus (GEO) with the accession number GSE208637,
originally generated by a study investigating the role of
CD161-expressing CD8+ T cells in chronic Hepatitis B (HBV) infection.
The study aimed to understand the profiles and changes of these cells
during the infection. Utilizing blood and liver tissue samples from
chronic HBV patients, the dataset distinguished between two groups based
on the severity of liver injury indicated by Aspartate Aminotransferase
(AST) levels---AST\>40 representing more severe liver injury and AST\<40
representing less severe injury. Figure 1 shows the difference in tissue
appearance between the 2 conditions and it is derived from the dataset's orginial paper @liu2023increased. All figures in the Background section are created using the knitr package (@R-knitr).

```{r, echo=FALSE, warning=FALSE, error=FALSE, out.width = "400px", fig.align='center', fig.cap="Figure 1: Comparison of Liver tissue from HBV infected patients without liver injury (AST<40) and with Liver injury (AST>40)"}
knitr::include_graphics("Figures/Tissue.png") 
```

This differentiation allowed for an analysis of the impact of liver
injury severity on CD161+CD8+ T cells. The dataset, which includes liver
tissues from HBV-infected patients, leverages next-generation sequencing
on the Illumina platform, offering comprehensive gene coverage. With 10
samples, including biological replicates for both conditions, the
dataset provided a robust basis for analysis. Below we can see a table
showcasing the samples in the dataset and the names used for them.

```{r, echo=FALSE, warning=FALSE, error=FALSE, out.width = "70%", fig.cap="Table 1: Summary of the Dataset Samples, their source and status and the short_name used to represent them" }
knitr::include_graphics("Figures/Samples\ Table.png") 
```

### Filtering & HGNC Mapping

The bioinformatics analysis began by filtering the RNAseq dataset to
enhance the quality of the data for differential expression analysis.
This was achieved by applying an edgeR-based filtering protocol, where
genes with less than 1 read per million in fewer than 4 samples (the
smallest number of replicates in a condition) were filtered out. This
step ensured the removal of low-abundance genes, which can introduce
noise and affect statistical power. This step reduced the number of
genes from 56275 to 16697.

Figure 3 below showcases the Boxplots and Density Plots for the
pre-filtered and post-filtered data. The boxplots show how removal of
low read counts lead to a more uniform distribution of expression levels
across samples and fewer outliers. Similarly the density plots which
show variation in gene expression levels across the samples changed to
become smoother and more consistent in distribution.

```{r, echo=FALSE, warning=FALSE, error=FALSE, out.width = "50%", fig.show='hold', fig.cap="Figure 2: Visualization of data pre and post filtering. Top Left = Pre-filtering Boxplot, Top right = Pre-filtering Density Plot, Bottom Left = Post-filtering boxplot, Bottom Right = Post-filtering Density Plot"}
knitr::include_graphics(c("Figures/Pre-filtering\ boxplot.png", "Figures/Pre-filtering\ density plot.png", "Figures/Post-filtering\ boxplot.png", "Figures/Post-filtering\ density plot.png"))
```

After filtering, gene names were mapped to their respective HGNC
symbols, ensuring that gene identifiers conformed to standard
nomenclature. Out of the original set, 475 non-approved gene symbols
were replaced with HGNC-approved symbols, while 2030 remained unmapped.

## Normalization

Next we Normalized the gene counts using Trimmed Mean of M-values
normalization method. The boxplot and density plot below showcase the
effective standardization of the data due to normalization. An MDS plot
also showed noticeable difference between samples of different
conditions based on their expression profile.

```{r, echo=FALSE, warning=FALSE, error=FALSE, out.width = "50%", fig.show='hold', fig.cap="Figure 3: Visualization of Normalized data. Top Left = Pre-filtering Boxplot, Top right = Pre-filtering Density Plot, Bottom Left = Post-filtering boxplot, Bottom Right = Post-filtering Density Plot"}
knitr::include_graphics(c("Figures/Normalized\ boxplot.png", "Figures/Normalized\ density plot.png", "Figures/MDS\ plot.png"))
```

# Differential Gene Expression

The first thing to get started with the Differential gene expression
analysis is to solidify the model we are working with. This will require
us to inspect the MDS plot to see what factors are important to
incorporate into the model. Estimating dispersion and carrying our the differential gene expression was done using the edgeR pacakge protocol (@R-edgeR)

## Model Design

Based on the MDS plot conducted in the previous section we can conclude
that the condition or presence of Liver injury (H/AST\>40 or L/AST\<40)
in the patient is the main factor to consider for the model design. This
is what I will be doing.

The first step will be to rerun the code from the previous assignment to
have the matrix of normalized data.

```{r, echo=FALSE, warning=FALSE, error=FALSE, include=FALSE}
# The accession id for the dataset
geo_accession_id <- "GSE208637"

# get the name of the supplementary file
supplmentary_files = GEOquery::getGEOSuppFiles(geo_accession_id, fetch_files = FALSE)
dataset_file_name <- supplmentary_files$fname

# Download the dataset

# save current directory
working_dir <- file.path(getwd())

# checking to see if the dataset file exists already to download file once and 
# avoid redownloading 
missing_files <- supplmentary_files$fname[!unlist(
  lapply(supplmentary_files$fname, FUN=function(x){
    file.exists( file.path(working_dir, geo_accession_id,x))}))]

# Downlod the file if it has not been downloaded 
if (length(missing_files) > 0){
  # Get the dataset
  supp_file = GEOquery::getGEOSuppFiles(GEO = geo_accession_id, 
                                        filter_regex = missing_files,
                                        baseDir = working_dir,
                                        fetch_files = TRUE)
}

# Create the relative path for the downloaded dataset file
dataset_path <- paste(working_dir,"/GSE208637/GSE208637_readcounts.txt.gz", sep = "")

# read the dataset into a table
liver_disease_data <- read.table(dataset_path, header = TRUE, check.names = TRUE)

# get the list of gene ids which are in the first column of the dataset 
dataset_rownames <- liver_disease_data[,1]

# create a matrix with the count values
dataset_matrix <- as.matrix(liver_disease_data[,2:11])

# set the row names for the matrix as the gene ids
row.names(dataset_matrix) <- dataset_rownames

# our knockdown number
knockdown_number <- 4

# calculate the counts per million for our data matrix
cpms <- edgeR::cpm(dataset_matrix)

# find the rows we want to keep
keep <- rowSums(cpms > 1) >= knockdown_number

# Create new counts datamatrix containing data after filtering
counts <- dataset_matrix[keep,]

# install HGNChelper package if not installed
if (!requireNamespace('HGNChelper', quietly = TRUE)) {
    install.packages('HGNChelper')
}

# get the gene names from the count matrix rownames
filtered_ds_rownames <- row.names(counts)

# Check if the gene names in my dataset are HGNC approved
checked_hgnc_symbols <- HGNChelper::checkGeneSymbols(filtered_ds_rownames)

not_approved_genes <- checked_hgnc_symbols[checked_hgnc_symbols$Approved == FALSE,]

# get only the  non-approved genes with a suggested replacement sumbol
rp_not_approved_genes <- checked_hgnc_symbols[checked_hgnc_symbols$Approved == FALSE &
                                                (!is.na(checked_hgnc_symbols$Suggested.Symbol)),]
# keep only the column with the original symbol and replacement symbol
rp_not_approved_genes <- rp_not_approved_genes[, c(1,3)]


# Loop through each row of 'rp_not_approved_genes'
for(i in 1:nrow(rp_not_approved_genes)) {
  # Get the original gene name and the replacement gene name
  original_gene_name <- rp_not_approved_genes[i, 1]
  replacement_gene_name <- rp_not_approved_genes[i, 2]
  
  # Find the index of the original gene name in the row names of 'counts'
  gene_index <- which(rownames(counts) == original_gene_name)
  
  # Replace the original gene name with the replacement gene name in 'counts'
  if(length(gene_index) > 0) { # Only replace if the original gene name is found
    rownames(counts)[gene_index] <- replacement_gene_name
  }
}

# create edgeR DGElist datatype of our data
edge_list <- edgeR::DGEList(counts = counts, group = colnames(counts))

# calculate the normalization factors
estimated_norm_factors = edgeR::calcNormFactors(edge_list)

# Create Normalized Counts
normalized_counts <- edgeR::cpm(estimated_norm_factors)
```

## Estimate Dispersion

Before carrying out the differential expression we create a DGEList
object to carry out and estimate the dispersion of the counts. This is
done to as estimating dispersion is a crucial step in RNA-Seq data
analysis for modeling the inherent variability in the data accurately.
This step ensures that the statistical tests used to identify
differentially expressed genes are based on realistic assumptions about
the data's behavior, leading to more reliable and interpretable results.
It is also required before carrying out differential expression using
the negative binomia model used by edgeR.

Moreover, we are setting the levels to be low_AST first meaning, we are
setting AST\<40 or No Liver injury to tbe the base to which we will
compare the expression in the high AST\>40 samples to.

```{r, echo=TRUE, warning=FALSE, error=FALSE}

# Create DGEList object for dispersion
dispersion_edge_list <- edgeR::DGEList(counts = counts)

# Create groupings from samples 
groups <- factor(c('low_AST', 'low_AST', 'low_AST', 'low_AST', 'high_AST', 'high_AST', 'high_AST', 'high_AST', 'high_AST', 'high_AST'), levels = c("low_AST", "high_AST"))
design <- model.matrix(~groups) 


# Update the DGEList object with the design matrix
dispersion_edge_list$design <- design

# use the edgeR package to estimate dispersion
dispersion_edge_list <- edgeR::estimateDisp(dispersion_edge_list, design)
```

## Carrying Out Differential Expression

To carry out differential expression we use the Quasi Liklihood fit
model as it is the best for Bulk RNAseq data. First we fit the data into
the QLfit model and then we run the gene expressoin analysis on it. We
set the coef as groupshigh_AST. This ensures that we are comparing the
control or AST\<40/No liver injury group which is taken as the default
to the AST\>40/Liver injury group. We then use topTags to extract the
results of the differential expression analysis. Kable was used to plot produce the table of results (@R-kableExtra), edgeR was used for coming up with the topTags (@R-edgeR).

```{r, echo=TRUE, warning=FALSE, error=FALSE}
# Fit data into Quasi Liklihood fit model
fit <- edgeR::glmQLFit(dispersion_edge_list, design)

# Run Differentil Gene Expression analyis on the 
qlf.low_vs_high <- edgeR::glmQLFTest(fit, coef='groupshigh_AST')

# Extract differential expression analysis results 
qlf_output_hits <- edgeR::topTags(qlf.low_vs_high, sort.by = "PValue", n = nrow(counts))

# Plot some of results
kableExtra::kable(edgeR::topTags(qlf.low_vs_high), type="html",row.names =TRUE, align="c")
```

**Table 2:** Above we see what the result from the differential
expression analysis looks like. logFC represents the log fold change so
the relative expression of the gene, logCPM is the relative counts per
million of the gene, PValue is the significance of the expression of
this gene, and FDR is the significance of expression of the genes after
correction. BH indicates that the Benjamni - Hochberg was the used
method for multiple hypothesis testing.

## Threshold Selection

When selecting a threshold we want to set one that filters out enough of
the genes so that we are only left with the top hits and most relevant
and differentially expressed genes. We will start with the high
recommended P-value of 0.05 and adjust accordingly. Below we see the
number of genes passed the intially set P-value Threshold

```{r, echo=TRUE, warning=FALSE, error=FALSE}
# Genes that pass the threshold p-value
sprintf("There are %d genes that pass the threshold p-value of 0.05", length(which(qlf_output_hits$table$PValue < 0.05)))

# Genes that pass correction with a threshold of 0.05
sprintf("There are %d genes that pass correction with a threshold of 0.05", length(which(qlf_output_hits$table$FDR < 0.05)))
```

Due to the large number of genes that pass the threshold, I set the threshold to be a
stringent 0.000001 to ensure only a couple hundred genes pass the
threshold and so that we only get the top and most important hits. Below
we see the number of genes that passed both the threshold p-value before
and after correction.

```{r, echo=TRUE, warning=FALSE, error=FALSE}
# Genes that pass the threshold p-value
sprintf("There are %d Genes that passed the more stringent threshold p-value of 0.000001 ", length(which(qlf_output_hits$table$PValue < 0.000001)))

# Genes that pass correction with a more stringent threshold of 0.000001
sprintf("There are %d Genes that passed correction with the more stringent threshold p-value of 0.000001",
length(which(qlf_output_hits$table$FDR < 0.000001)))
```

This is a good enough value for the number of genes passing correction so
we are going with this threshold value.

## MA plot

To visualize the differential gene expression between the the Low
AST\<40 and the high AST\>40 group we us an MD plot as seen below. This
plot plots the Average Log of the counts per million (CPM) on the x-axis
and the Log Fold Cchange (LogFC) on the y-axis. LogCPM shows us the the
average expression level of genes across the two conditions and LogCF
shows us how much more or less a gene is expressed in the high AST\>40
condition compared to the Low AST\<40 condition. By setting the
threshold value to be our previously chosen 0.000001 we can highlight
the genes that were up regulated and downregulated. Genes that were
upregulated are those that pass correction with for our threshold value
and have a LogFC value \> 0. Genes that were down regulated are those
that pass correction with for our threshold value and have a LogFC value
\< 0. MA plot created using edgeR (@R-edgeR)

```{r, echo=TRUE, warning=FALSE, error=FALSE}
edgeR::plotMD.DGELRT(qlf.low_vs_high, xlab="Average log CPM", ylab="log-fold-change (Low_AST vs High_AST)", 
       main="MA plot", status=NULL, 
       p.value=0.000001)
abline(h = 0, col="green")
```

**Figure 4:** MA plot of the Differential gene expression analysis
between the Low AST\<40 condition and the High AST\>40 condition.
Upregulated significant genes are red and down regulated significant
genes are blue.

## Heat Map

To visualize represent the expression patters of the genes across the
samples and conditions we create a heatmap of normalized counts for
significant genes that passed your correction threshold.Theatmap of the
significant genes provides a comprehensive overview of gene expression
differences between high and low AST conditions, offers insights into
the biological significance of these differences, and helps in
identifying patterns that might be relevant for understanding the
condition or disease under study. Below we can see significant
difference in expression of genes between the samples with almost no
overlap of expression. This shows us noticeable gene profiles in the two
tissue types. 
Heat maps created using ComplexHeatmap package (@ComplexHeatmap_2022). Heatmap matrix annotated using circlize package (Circlize_2014).

```{r, echo=TRUE, warning=FALSE, error=FALSE}
# Get the dataframe with the expression results for the signnificant genes
significant_genes_df <- qlf_output_hits$table[qlf_output_hits$table$FDR < 0.000001, ]

# extract the significant gene names
significant_genes <- row.names(significant_genes_df)

# keep the normalized counts for the significant genes only
significant_counts <- normalized_counts[unlist(significant_genes), ]

# install packages if not installed
if (!requireNamespace('ComplexHeatmap', quietly = TRUE)) {
    install.packages('ComplexHeatmap')
}
if (!requireNamespace('circlize', quietly = TRUE)) {
    install.packages('circlize')
}

# Scale the counts data so that down regulated genes are blue and upregulated genes are red providing greater contrast for comparison of expression
heatmap_matrix <- t(scale(t(significant_counts)))

# Set colours for expression difference 
if(min(heatmap_matrix) == 0){
  heatmap_col = circlize::colorRamp2(c( 0, max(heatmap_matrix)),
                           c( "white", "red"))
  } else {
    heatmap_col = circlize::colorRamp2(c(min(heatmap_matrix), 0,
                               max(heatmap_matrix)), c("blue", "white", "red"))
  }

# set colours to represent samples from each condition
unique_ast_level_colors = rainbow(n=2)
names(unique_ast_level_colors) <- c("low_AST", "high_AST")

# create annotations with the chosen colours
ha_pat <- ComplexHeatmap::HeatmapAnnotation(df = data.frame(
  Ast_level = c('low_AST', 'low_AST', 'low_AST', 'low_AST', 'high_AST', 'high_AST', 'high_AST', 'high_AST', 'high_AST', 'high_AST')))

# Plot heatmap
current_heatmap <- ComplexHeatmap::Heatmap(as.matrix(heatmap_matrix),
                           top_annotation = ha_pat,
                           show_row_dend = TRUE,show_column_dend = TRUE,
                           col=heatmap_col,show_column_names = TRUE,
                           show_row_names = FALSE,show_heatmap_legend = TRUE,
                           column_title =("Top hits L/Low AST vs H/High AST"))
current_heatmap
```

**Figure 5:** The heatmap showcasing the difference in expression
between the two conditions (Low AST<40 and High AST>40), and the samples L1-L4 and H1-H6.

# Thresholded over-representation analysis

Next we carried out Thresholded over-representation analysis or Gene Set
Enrichment Analysis (GSEA). This analysis is conducted to help us
identify biological processes, molecular functions, and cellular
components that are significantly enriched in our gene list. This means
it reveals what processes or functions are overrepresented in our
dataset, suggesting the biological themes that our genes are involved
in.

## Upregulated and Down Regulated genes

By looking at the genes that pass our previously set threshold for
correction and looking at whether their logFC or Log Fold Count values
above or below 0, we can know which genes of interest are upregulated or
down regulated respectively.

```{r, echo=TRUE, warning=FALSE, error=FALSE}
# Number of upregulated Genes
sprintf("There are %d upregulated genes", 
        length(which(significant_genes_df$logFC > 0)))

# Number of down-regulated Genes
sprintf("There are %d down-regulated genes",
        length(which(significant_genes_df$logFC < 0)))
```

## Selecting GSEA tool

To carry out the Gene Set Enrichment, I chose the G:Profiler tool.
G:Profiler was the recommended and ideal choice for this project as it
has a wide range of gene sets. It also has an accessible easy to use r
package that is updated regularly with the last update being last month.

## Extracting differently regulated genes

The first thing to do is to create sets of for the significant genes,
upregulated significant genes and down regulated significant genes. This
will allow us to then run the enrichment analysis on the whole set of
genes and then on the differently regulated genes seperately.

```{r, echo=TRUE, warning=FALSE, error=FALSE}
significant_genes <- row.names(significant_genes_df)

upregulated_significant_genes <- row.names(subset(significant_genes_df, logFC > 0))

down_regulated_significant_genes <- row.names(subset(significant_genes_df, logFC < 0))
```

## Carrying Out The Enrichment Analysis

To conduct the gene set enrichment analysis the gprofiler2 package was
used (cite). I ran the query on the whole list of significant genes before
restricting the size of gene sets in the results and after restircting
the size of the gene sets in the results. The significant argument was
set to TRUE so that only gene sets that meet our set threshold will be
returned. Ordered Query is set to false as the query list is not
ordered. I used our previous threshold of 0.000001

### Annotation sources

For the annotation sources I used GO biological process (<GO:BP>),
WikiPathways (WP) and Reactome (Reac). I used them as they are excellent
choices for gene enrichment analysis due to their comprehensive
coverage, detailed annotation, and widespread recognition within the
scientific community. They are all frequently updated making them
reliable sources. The integration of results across these three
databases enhances the reliability of the findings and provides a robust
framework for interpreting the biological implications of genomic data.

### Non-size Restricted All significant Genes GSEA

Below is the code for the gene enrichment analysis for the whole gene
list. The results of the gene enrichment can be seen in the Table below.
This Table shows us the term name of the top hits from gene sets by
P-value. This shows the most relevant gene sets and the pathways or
biological mechanisms they are involved in. The table is accompanied by
a plot of the gene enrichment results showing the number of gene sets in
the results for each data source. The number of genes in each gene set
is represented by the size of the dot representing that gene set so a
bigger dot is a larger sized gene set.

Tables are produced using the kable package (@R-kableExtra) and plot of enrichment analysis created using gprofiler2 package (@gprofiler2_2020)
```{r, echo=TRUE, warning=FALSE, error=FALSE}
# Ensure the gprofiler2 package is installed
if (!requireNamespace("gprofiler2", quietly = TRUE)) {
  install.packages("gprofiler2")
}

# Conduct gene enrichment analysis
whole_gene_results <- gprofiler2::gost(query = significant_genes,
                     organism = "hsapiens", 
                     significant= TRUE,
                     ordered_query = FALSE,
                     exclude_iea=TRUE,
                     sources = c("GO:BP", "WP", "REAC"),  # GO BP, WikiPathways, Reactome
                     correction_method = "fdr",
                     user_threshold = 0.000001)  

sprintf("There are %s gene sets in the Non-size restricted All genes GSEA results", length(row.names(whole_gene_results$result)))

# Show table of results 
kableExtra::kable(whole_gene_results$result[, c("term_name", "p_value", "term_size", "source")], type="html", row.names =FALSE, digits = 12, padding="300", align="c")

```

**Table 3:** The above table showcases the Term names, term_sizes or gene set sizes, P-value associated with the gene set and source of the gene set for the All significant genes GSEA results before size restriction.

```{r, echo=TRUE, warning=FALSE, error=FALSE}
# plot enrichment whole gene list enrichment analysis 
enrich_plot <- gprofiler2::gostplot(whole_gene_results, capped = FALSE, interactive = FALSE)

# add a title
enrich_plot_with_title <- enrich_plot + ggplot2::ggtitle("All significant Genes GSEA")
enrich_plot_with_title
```

**Figure 6:** The above figure show the plot of the Gene Set Enrichment Analysis results. The dot colours signify which source the gene sets are coming from. The size of the dots represent the size of the gene sets.

### Size Restricted All significant Genes GSEA

Despite the initial number of gene set results not being particularly
high, I decided to apply restrictions on the size of these gene sets for
several reasons. Firstly, focusing on gene sets of moderate size
enhances the statistical robustness and biological relevance of our
findings. This approach aids in avoiding potential overinterpretation of
results driven by very small gene sets, which may lack statistical
confidence, and overly broad interpretations from excessively large gene
sets, which can dilute specific biological insights. Still I set the Max
size restriction to be 800 as the significance threshold was really
stringent and so we actually have some gene sets in the results. The Min
size restriction was set to 5.

```{r, echo=TRUE, warning=FALSE, error=FALSE}

# restrict the size of the results 
size_restricted_results <- whole_gene_results$result[which(whole_gene_results$result$term_size >=5 & whole_gene_results$result$term_size <=800),]

whole_gene_results$result <- size_restricted_results

sprintf("There are %s gene sets in the size restricted All genes GSEA results", length(row.names(whole_gene_results$result)))

# Show table of results 
kableExtra::kable(whole_gene_results$result[, c("term_name", "p_value", "term_size", "source")], type="html",row.names =FALSE, digits = 12, padding="300", align="c")

```

**Table 4:** The above table showcases the Term names, term_sizes or gene set sizes, P-value associated with the gene set and source of the gene set for the size restricted All genes GSEA results.

Due to size restriction the number of gene sets in the whole significant
gene list GSEA went from 7 to 6 removing very broad biological pathways.
The first was called small molecule metabolic process and the second was
called metabolism which we can see are veyr broad and vague term names.
This is what we expected and wanted from restricting gene set size

## Up and Down Regulated Genes GSEA

Next we ran the Gene set enrichment analysis on the upregulated and the
down regulated list of significant genes separately. This allowed to see
which biological pathways and mechanisms specifically were most relevant
to the High AST\>40 condition and which were most relevant to the Low
AST\<40 condition. We went with the restricted gene set size results as
the default here.

### Upregulated GSEA

Below is the code for the upregulated genes GSEA. After it we display a table to showcase the top 10 gene sets by significance in P-value. The term_name lets us know the most significant and relevant biological pathways in the upregulated GSEA results. 

```{r, echo=TRUE, warning=FALSE, error=FALSE}
# Conduct gene enrichment analysis
upregulated_genes_results <- gprofiler2::gost(query = upregulated_significant_genes,
                     organism = "hsapiens", 
                     significant=TRUE,
                     ordered_query = FALSE,
                     exclude_iea=TRUE,
                     sources = c("GO:BP", "WP", "REAC"),  # GO BP, WikiPathways, Reactome
                     correction_method = "fdr",
                     user_threshold = 0.000001)  


size_restricted_results <- upregulated_genes_results$result[which(upregulated_genes_results$result$term_size >=5 & upregulated_genes_results$result$term_size <=800),]

upregulated_genes_results$result <- size_restricted_results

sprintf("There are %s gene sets in the upregulated genes GSEA results", length(row.names(upregulated_genes_results$result)))

# Show table of results 
kableExtra::kable(upregulated_genes_results$result[1:10, c("term_name", "p_value", "term_size", "source")], type="html",row.names =FALSE, digits = 12, padding="300", align="c")

```

**Table 5: ** The above table showcases the Term names, term_sizes or gene set sizes, P-value associated with the gene set and source of the gene set for the size upregulated genes GSEA results

### Down Regulated GSEA

Below is the code for the down regulated genes GSEA. After it we display a table to showcase the top 10 gene sets by significance in P-value. The term_name lets us know the most significant and relevant biological pathways in the down regulated GSEA results. 

```{r, echo=TRUE, warning=FALSE, error=FALSE}
# Conduct gene enrichment analysis
down_regulated_genes_results <- gprofiler2::gost(query = down_regulated_significant_genes,
                     organism = "hsapiens", 
                     significant= TRUE,
                     ordered_query = FALSE,
                     exclude_iea=TRUE,
                     sources = c("GO:BP", "WP", "REAC"),  # GO BP, WikiPathways, Reactome
                     correction_method = "fdr",
                     user_threshold = 0.000001)  

size_restricted_results <- down_regulated_genes_results$result[which(down_regulated_genes_results$result$term_size >=5 & down_regulated_genes_results$result$term_size <=800),]

down_regulated_genes_results$result <- size_restricted_results

sprintf("There are %s gene sets in the down regulated genes GSEA results", length(row.names(down_regulated_genes_results$result)))

# Show table of results 
kableExtra::kable(down_regulated_genes_results$result[1:10, c("term_name", "p_value", "term_size", "source")], type="html",row.names =FALSE, digits = 21, padding="300", align="c")

```

**Table 6:** The above table showcases the Term names, term_sizes or gene set sizes, P-value associated with the gene set and source of the gene set for the size down regulated genes GSEA results

## Analysis Comparison (Whole Gene List vs upregulated vs downregulated)

The Gene Set Enrichment Analysis (GSEA) provides a distinct overview of the biological processes implicated in High AST>40 versus Low AST<40 conditions through the evaluation of all significant genes and the subsets of upregulated and downregulated genes. 

When looking at the overall significant genes the GSEA identifies 6 gene sets. However, it identifies 20+ gene sets for both the up and down regulated genes GSEA results. So quite a smaller number of gene sets in the results of the overall significant genes GSEA

For the upregulated genes in the High AST>40 samples, the GSEA reveals enrichment in immune system-related processes, including B cell mediated immunity and complement system activation. This suggests a heightened immune response or activation of immune pathways in the context of higher AST levels.

In contrast, the downregulated genes in the High AST>40 samples exhibit enrichment in a variety of metabolic processes, including organic acid, carboxylic acid, and amino acid metabolic processes. The downregulation of these genes points to a potential suppression or disruption of metabolic pathways in patients with higher AST levels.

Finally, the narrower range of biological processes for the overall significant genes GSEA are primarily metabolic in nature. This highlights key metabolic activities that may be broadly affected by AST levels but does not show the immune component that distinguishes the upregulated genes.


# Interpretation

## Do the over-representation results support conclusions or mechanism discussed in the original paper?

Yes, the over-representation results do support the conslusions that the authors of the original paper reached. They mainly focused on the upregulation of immune response related genes such as IL1Î² and IL8 which play roles in chemotaxis and activation of neutrophils (@liu2023increased). Although these genes were removed when we made our threshold more stringent to get a lower number of genes that pass correction, the over-representation results for the upregulated genes show that most of the top hits have do with the immune system and immune responses as a whole. 

## Supporting Evidence

This is supported by papers such as that by Polyak et al. (@polyak2001elevated) that show that il8 gene plays a role in the recruiting of neutrophils (an immune pathways related response) to exacerbate inflammation-mediated liver injury. Thus, are results are correct in suggesting that the upregulated genes in the High AST>40 group of samples would have the most relevant biological pathways or gene sets be.

# References 
